source/grid/grid_in.cc

```diff
@@ -340,139 +340,181 @@ GridIn<dim, spacedim>::read_vtk(std::istream &in)
       for (unsigned int i = 0; i < n_ints; ++i)
         in >> tmp_int;
 
-      // Ignore everything up to CELL_DATA
-      while (in >> keyword)
-        if (keyword == "CELL_DATA")
+      /////////////////////Processing the CELL_DATA
+      /// section////////////////////////
+      in >> keyword;
+
+      AssertThrow(
+        keyword == "CELL_DATA",
+        ExcMessage(std::string(
+          "While reading VTK file, missing CELL_DATA section. Found <" +  
```

Comment:
This implies that *every* file you want to read in now has to have a `CELL_DATA` section where before it was optional. Can you explain why this is now necessary?

Reply:
I've tried to read the part below that corresponds to the existing code. It's a bit complicated to read because the indentation changed, but I don't actually see that you changed anything in the part that reads the cell data -- or did you?